pandas/tests/groupby/test_categorical.py

```diff
@@ -1193,3 +1193,15 @@ def test_groupby_categorical_axis_1(code):
     result = df.groupby(cat, axis=1).mean()
     expected = df.T.groupby(cat, axis=0).mean().T
     assert_frame_equal(result, expected)
+
+
+def test_groupby_cat_preserves_group_name():
+    # GH 28787
```

Comment:
can you use the example from the original issue and compare the resultant dataframes. I don't like *just* checking the names; rather a full-on comparison is best.